server/service/async/async.go

```diff
@@ -89,14 +88,36 @@ func (t *Task) RecordLabelQueryExecutions(ctx context.Context, host *fleet.Host,
 	keySet := fmt.Sprintf(labelMembershipHostKey, host.ID)
 	keyTs := fmt.Sprintf(labelMembershipReportedKey, host.ID)
 
+	// set an expiration on both keys (set and ts), ensuring that a deleted host
+	// (eventually) does not use any redis space. Ensure that TTL is reasonably
+	// big to avoid deleting information that hasn't been collected yet - 1 week
+	// or 10 * the collector interval, whichever is biggest.
+	//
+	// This means that it will only expire if that host hasn't reported labels
+	// during that (TTL) time (each time it does report, the TTL is reset), and
+	// the collector will have plenty of time to run (multiple times) to try to
+	// persist all the data in mysql.
+	ttl := labelMembershipKeysMinTTL
+	if maxTTL := 10 * t.CollectorInterval; maxTTL > ttl {
+		ttl = maxTTL
+	}
+
+	// keys and arguments passed to the script are:
+	// KEYS[1]: keySet (labelMembershipHostKey)
+	// KEYS[2]: keyTs  (labelMembershipReportedKey)
```

Comment:
How do we know these two keys are in the same node?

Reply:
That's a very good question - this is due to the "hashtag" that is present in those keys (https://github.com/fleetdm/fleet/pull/3639/files/ece8828b02dea9483846f74ca519d0132013b816#diff-7cf9b593940195e55552c44ac2ca78ea746e5bf5311e2ab5980394a7967c16e2R18-R19), the `{%d}` where `%d` is the host id means that redis will hash just that part of the key to identify the node, and so both keys for a given host are guaranteed to live on the same node.

Reply:
Oh cool! (I asked because I reminded what you mentioned around redis scripts)